Makefile

```diff
@@ -0,0 +1,116 @@
+# run makefile from repo root
+
+.PHONY: build
+build:
+	GO_BUILD_FLAGS="-v" ./build
+	./bin/etcd --version
+	ETCDCTL_API=3 ./bin/etcdctl version
+
+# run all tests
+test-all:
+	RELEASE_TEST=y INTEGRATION=y PASSES='build unit release integration_e2e functional' ./test 2>&1 | tee test.log
+
+# clean up failed tests, logs, dependencies
+clean:
+	rm -f ./*.log
+	rm -f ./bin/Dockerfile-release
+	rm -rf ./gopath
+	rm -rf ./release
+	rm -f ./integration/127.0.0.1:* ./integration/localhost:*
+	rm -f ./clientv3/integration/127.0.0.1:* ./clientv3/integration/localhost:*
+	rm -f ./clientv3/ordering/127.0.0.1:* ./clientv3/ordering/localhost:*
+
+# keep in-sync with 'Dockerfile-test', 'e2e/docker-dns/Dockerfile'
+_GO_VERSION = go1.9.1
+ifdef GO_VERSION
+	_GO_VERSION = $(GO_VERSION)
+endif
+
+# build base container image for testing on Linux
+docker-test-build:
+	docker build --tag gcr.io/etcd-development/etcd-test:$(_GO_VERSION) --file ./Dockerfile-test .
```

Comment:
You'll need to have some way to distinguish the docker image architecture.  Maybe something like `gcr.io/etcd-development/etcd-test:$(_GO_VERSION)-$(ARCH)`., and also have arch specific docker files.